contracts/BodhiCandy.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity ^0.4.18;
+
+import './libs/SafeMath.sol';
+
+contract BodhiCandy {
+    using SafeMath for uint256;
+
+    // Amount in Satoshi to deposit
+    uint256 public constant depositAmount = 1000000;
+
+    // Amount of blocks it takes for the last person to deposit to win if no others deposit within the length.
+    uint256 public constant winningBlockLength = 30;
+
+    address public lastDepositer;
+    uint256 public lastDepositBlock;
+    uint256 public currentBalance;
+
+    event UserWon(address indexed winner, uint256 amountWon);
+    event Returned(uint256 returned);
+
+    function deposit() external payable {
+        require(msg.value >= depositAmount);
+
+        // Calculate any amount over the depositAmount to return to user
+        uint256 excessDeposit = 0;
+        if (msg.value > depositAmount) {
+            excessDeposit = msg.value.sub(depositAmount);
+        }
+
+        // Last depositer wins
+        if (lastDepositer != address(0) 
+            && lastDepositBlock != 0 
+            && block.number.sub(lastDepositBlock) >= winningBlockLength) {
+
+            uint256 amountWon = currentBalance.add(excessDeposit);
+            currentBalance = 0;
+
+            if (amountWon > 0) {
```

Comment:
Do you need this if checking? I think amountWon is provable always > 0.

Reply:
This was to prevent reentrancy transferring. In the case where the reentrant calls the contract twice, after the first time through, the currentBalance will be 0. So if the second time through, if the attacker calls it again with no excessDeposit, then amountWon would be 0.

But I just found a bug where if the attacker sent excess qtum (excessDeposit > 0), the attacker can withdraw again since amountWon will be 0.

I am going to add a check for currentBalance > 0 in the first if requirements.